### ISSUE TITLE
WEP security not supported with Photon

### ISSUE THREAD

**Issue by karlhenricksen** [spark/spark-setup-ios#30]
Hi, I have only tried one router that supports WEP-security type so far, Linksys wrt-54g, but when I set that up and attempt to set Photon wifi up on that router using the particle app, it hangs at the "Connect to Wi-Fi network" screen for 2 min, before it throws up the "Oops" screen 3-7 times (this is a secondary issue with the app). Resetting the photon results in a solid / slightly flickering green LED and is unusable. Note there were no issues with connecting to this router with other devices, so I ruled out incorrect login/setup on phones and laptops.

We also see the same failure when we update wifi creds via serial

Steps:
1) set up a router with WEP security - 64 or 128 bit (copy the WEP key)
2) set Photon into blue blinking wifi setup mode
3) connect phone to WEP router using WEP key
4) launch Particle wifi app and walk through steps 
- connect to Photon-xxxx
- enter WEP router credentials

Results:
2 min of waiting for Photon to connect followed by 3-7 rapid flashes of the "Oops" screen, then Photon is in steady/flickering green LED mode

HW/SW: 
Photon system v 0.4.5
Router: Linksys wrt-54GL - WEP 64/128bit
Phone: iOS9

![particle_connecting](https://cloud.githubusercontent.com/assets/12628878/10147939/d572218c-65fe-11e5-84be-35a9814340e9.PNG)
![particle_oops](https://cloud.githubusercontent.com/assets/12628878/10147940/d573140c-65fe-11e5-9df7-d3d9e006d715.PNG)

**Comment by karlhenricksen** [comment]
Also, should I open a separate issue for the multiple "Oops" screens that pop up on the app? Would imagine this may be one of the only times you'd see that problem, but it's not pretty.

**Comment by m-mcgowan** [comment]
@karlhenricksen have you seen this - https://community.particle.io/t/how-to-setup-wep-secured-network-on-the-photon/16026. Does formatting the password help?

**Comment by karlhenricksen** [comment]
@m-mcgowan yes, that does help thanks, it appears to go through successfully though I do have to reset every time afterwards (FYI https://github.com/spark/firmware/issues/620 - setup with WEP looks to be a consistent way to reproduce that other issue). 

Unfortunately this isn't a great solution for the customers that still use WEP (or have no control over their internet connections) since they wouldn't know to do this or have any feedback why their key doesn't work for Photon (though it's fine for other devices), so maybe it's more of an app issue - couldn't that be handled there? Just looking for a reasonable customer solution on this.

**Comment by pomplesiegel** [comment]
Hey @m-mcgowan, would it then make sense to have `spark-setup-ios` add the additional WEP formatting to the customer's inputted WEP password?

When a customer enters a WEP password for their computer or smartphone to join a network, they need not add any additional formatting. Shouldn't our WiFi migration app work in the same way?

**Comment by idokleinman** [comment]
@pomplesiegel I guess we can make the device setup lib format the password for WEP cases. It is not high priority on our backlog tasks currently. You're most welcome to add this code and send a PR.
